Remove NYA companies, add blog post

theymos opened the pull request:
In accordance with my proposed plan in #1824, this removes all of the NYA companies and adds a blog post warning about a future alert. I used @harding's version of the blog post, but modified it to be more specific. Currently, the deadline is set to Oct 11, though that can be changed, of course.

Perhaps non-obviously, as part of removing blockchain.info, I changed the pricing JavaScript from using bc.i to using bitcoinaverage.com.

achow101 commented:
What about BitGo? I know they didn't sign the agreement, but their CEO seems to be pushing really hard for Segwit2x, although it seems like a lot of people within the company don't want it.

achow101 commented on _posts/2017-10-05-denounce-segwit2x.md:
```diff
@@ -10,7 +10,13 @@ date: 2017-10-05
+In particular, we need to know that:
+
+1. The company will not under any circumstances list "Segwit2x" as "BTC" and/or "Bitcoin". Note that [Bitcoin is not ruled by miners](https://en.bitcoin.it/wiki/Bitcoin_is_not_ruled_by_miners), and miner actions cannot be used as a justification to redefine Bitcoin.
+2. The company will not by default *convert* user BTC to S2X-coins. Distributing "air-dropped" S2X-coins to users is acceptable, however.
```
Can you clarify what is meant by "convert"? Do you mean switching to using S2X by default or selling all user BTC for S2X on an exchange somewhere?

luke-jr commented on _posts/2017-10-05-denounce-segwit2x.md:
```diff
@@ -0,0 +1,75 @@
+lang: en
+category: blog
+
+title: "Bitcoin.org to denounce Segwit2x"
```
Prefer if it's called simply 2X, since it has nothing to do with Segwit. Calling it "Segwit2x" plays into their deception.

luke-jr commented on _posts/2017-10-05-denounce-segwit2x.md:
```diff
@@ -0,0 +1,75 @@
+permalink: /en/posts/denounce-segwit2x.html
+date: 2017-10-05
+---
+On 2017-10-11 at noon (UTC), Bitcoin.org is planning to publish a banner on every page of the site warning users about the risks of using services that will default to the so-called Segwit2x (S2X) contentious hard fork. S2X companies will be called out by name. To ensure that we only warn users against companies that will actually put user deposits at risk, we urge all companies to publicly clari ...
```
2X is an altcoin, not a hardfork.

theymos replied:
I think that using that language just makes it too confusing in this case. "Contentious hardfork" and "altcoin" are pretty similar.

luke-jr replied:
Perhaps "contentious hardfork attempt"?

luke-jr commented on _posts/2017-10-05-denounce-segwit2x.md:
```diff
@@ -0,0 +1,75 @@
+
+Although bitcoin.org [condemns](/en/posts/hard-fork-policy) contentious hard fork attempts such as S2X, we consider it *tolerable* for companies to support S2X in ways that do not contradict the above three points, such as by supporting both Bitcoin and S2X simultaneously as separate cryptocurrencies.
+
+By default, we will be using the following list of companies known to support S2X in our warning:
```
Suggest explicitly reserving the right to add other companies that behave in a manner hostile to Bitcoin.

fraggle222 replied:
bitcoin.org should focus on technical issues, not corporate actions.

achow101 replied:
> bitcoin.org should focus on technical issues, not corporate actions.

I disagree. Read the [Bitcoin.org's mission](https://bitcoin.org/en/about-us#mission). I think that this clearly falls under

> Display transparent alerts and events regarding the Bitcoin network.

and being able to add and remove companies from that list enables us to be more informative in this alert regardingthe Bitcoin network.

Cobra-Bitcoin commented:
ACK. Once we get a few more ACK's, let's try to merge this in as soon as we can. Good catch by the way with the pricing JS. Completely forgot we were using blockchain.info there.

fraggle222 commented on _posts/2017-10-05-denounce-segwit2x.md:
```diff
@@ -0,0 +1,75 @@
+
+1. The company will not under any circumstances list "Segwit2x" as "BTC" and/or "Bitcoin". Note that [Bitcoin is not ruled by miners](https://en.bitcoin.it/wiki/Bitcoin_is_not_ruled_by_miners), and miner actions cannot be used as a justification to redefine Bitcoin.
+2. The company will not by default *convert* user BTC to S2X-coins. Distributing "air-dropped" S2X-coins to users is acceptable, however.
+3. The company will continue to provide normal service to Bitcoin (ie. non-SegWit2x) users.
```
Why the vague language with "air-dropped" ? This term is used in different ways in the industry and poorly understood. Discussion starts here https://bitcointalk.org/index.php?topic=1438371.0 . 

Distribution of coins (or providing means to safely move those coins) gained by owning the private keys for utxos before a fork is different than someone buying up bitcoin (or using bitcoin gained from an ico or other sale) and distributing it.

fraggle222 commented on _posts/2017-10-05-denounce-segwit2x.md:
```diff
@@ -0,0 +1,75 @@
+In particular, we need to know that:
+
+1. The company will not under any circumstances list "Segwit2x" as "BTC" and/or "Bitcoin". Note that [Bitcoin is not ruled by miners](https://en.bitcoin.it/wiki/Bitcoin_is_not_ruled_by_miners), and miner actions cannot be used as a justification to redefine Bitcoin.
+2. The company will not by default *convert* user BTC to S2X-coins. Distributing "air-dropped" S2X-coins to users is acceptable, however.
```
Suggest removing reference to "bitcoin is not ruled by miners". It adds little to this discussion and the wiki article doesn't really address the current situation fully.

harding replied:
I strongly support keeping it.  I think it's a very valuable contribution to this blog post as the NYA cited hash rate support as being necessary (though it wasn't), the segwit signaling part of btc1 required hash rate support (though BIP148 would've activated either way), and S2X supporters frequently point to the current "NYA" signaling of 90%+ of hash rate as heralding their eventual success (though it doesn't).

fraggle222 replied:
BIP148 itself was controversial and any unilateral uasf like that in the future will also be controversial. Suggesting that users stay on a chain with minimal hashpower can put their coins at risk. The article does not really address the situation of when part of the community wants to change consensus and another part does not except to assume that the part that does want to change the rules is an attack. This is not always the case. 

Linking bitcoin.org to wiki's that are subject to change and editing in general is a bad idea in general.

harding replied:
1. I agree BIP148 was controversial (I [did not support it](https://twitter.com/hrdng/status/869360246501847042) and I wrote the Bitcoin.org alert about it/BIP91).
2. The Wiki article does not address protocol upgrades because the Wiki article is not about protocol upgrades.  However, this Bitcoin.org post does link to a [previous Bitcoin.org post](https://bitcoin.org/en/posts/hard-fork-policy) describing what we think are inappropriate hard forks (any hard fork that is contentious).
3. It seems to me that for the past several years, the Wiki seems has been well maintained.  I added the article to my wiki watchlist which I have setup to email me (via my RSS reader) on any changes, so I can help police the article.  If it gets excessively vandalized, we can link to a specific revision.

fraggle222 replied:
Suggest you just link to the contentious hard fork post then. Although of course it does not define what 'contentious' is or what would not be contentious so it too is lacking but I think more on point to the current situation.

It is rather odd to link to a post like "bitcoin is not ruled by miners" in response to activity by miners. If it is not activity by miners but the corporate entities supporting NYA, then the post completely misses the mark.

I assume the intention is as a reminder to corporations and users that while they are free to run whatever software they want, they should not choose that software based on hash power alone. However, this blog post we are discussing (this pr) is specifically about services and corporations. Services and corporations are not miners. They choose what software to run as suits them. We cannot get inside their heads and say " oh you are doing it just because of hash rate". In fact we as a community are well aware that many services and corps support NYA for other reasons.

achow101 replied:
> Suggest you just link to the contentious hard fork post then.

It is already linked later in the post.

fraggle222 commented on _posts/2017-10-05-denounce-segwit2x.md:
```diff
@@ -0,0 +1,75 @@
+
+Although bitcoin.org [condemns](/en/posts/hard-fork-policy) contentious hard fork attempts such as S2X, we consider it *tolerable* for companies to support S2X in ways that do not contradict the above three points, such as by supporting both Bitcoin and S2X simultaneously as separate cryptocurrencies.
+
+By default, we will be using the following list of companies known to support S2X in our warning:
```
Suggest removing entire list of companies. It is impossible for bitcoin.org to keep track of each and every companies policies or intentions which can change at any time. Warn users of risks and why they exist.

harding replied:
This post is about warning companies and giving them a chance to leave the NYA before a list is published in an alert, so I think it's very useful to explicitly name the companies so they know they're on the list.

These companies previously publicly signed their name in support of the NYA, so I don't see why we should be timid about republishing that information.

fraggle222 replied:
Bitcoin.org should either develop objective rules for (similar to what is done for wallets) for services and corps, or leave well enough alone.

Bitcoin.org should not be about shaming corporations.

harding replied:
IMO, this list is not about shaming companies (how can you shame a corporate entity anyway?  corporations don't have feelings) but rather preparing to alert users about the danger of using businesses that may destroy their bitcoins.

"Previously signed the NYA and has not withdrawn" seems like an objective enough criteria to me.

Furthermore, it seems to me like you simply don't like the idea of Bitcoin.org publishing a statement about this subject at all.  You have a right to that opinion, but perhaps it would be better for you to state your opposition directly rather than trying to nitpick it to death.

fraggle222 replied:
I don't like the idea about calling companies out by name without specific risks cited that pertain exactly to that company's policy (we don't know how any of these companies will really handle 2x, how users funds will be at risk or not) and not offering users clear examples of risks and or benefits. I also don't like the idea of using bitcoin.org as some sort of  threat mechanism which this post clearly is.

I had hoped to try to salvage this and perhaps the "banner" that will follow, but it is begging to look unlikely that this is possible.

achow101 replied:
The whole point is to name specific companies so that they a) clarify their stance on 2X and b) inform users of specific companies that they might want to avoid because those companies are likely to switch users from using Bitcoin to S2X-altcoin.

harding commented:
Lightly-tested ACK.  For anyone else who wants to test, I have a preview here: http://dg4.dtrt.org/ (c124fb2b6f63419b9713a379d4708ca9c95d6bc2)

achow101 commented:
ACK

theymos commented:
@achow101 BitGo does seem suspicious, but there isn't enough solid info to list them right now IMO.

I changed the blog post to address some comments:

* I added quotes to the blog post title to indicate even more that Segwit2x is a deceptive name: 'Bitcoin.org to denounce "Segwit2x"'. And I added a footnote about this, as well.
* I changed the point about not converting BTC to be more specific.

Cobra-Bitcoin commented:
Merged! 👍

jnewbery commented:
This approach is unnecessarily divisive.

Did you speak to all of these companies _before_ "denouncing" and delisting them? Do you know what their understanding of the NYA was when they signed up? Do you know what their customers' needs are? Do you understand what their responsibilities are to their investors, their employees and their customers?

Of course you haven't, and yet you're happy to use your platform to publicly attack them.

Much more productive would be to issue a blanket warning to all Bitcoin users to understand what the hardfork in November means for their coins, and strongly advise them to contact all of their Bitcoin service providers and demand that their Bitcoin continue to be available and named Bitcoin after November. If you wanted to be really helpful you could translate that warning to Spanish and Swedish and Swahili and whatever other languages the users of these services speak.

And anyway, who made @theymos and @Cobra-Bitcoin the moral arbiters of Bitcoin? The previous PR (#1824) had NACKs from @wbnns, @ekerstein, @Mirobit, @nvk, @sunnankar and others. Why was this PR merged a day after opening without seeking feedback from people who had commented on the previous PR?

If you want to denounce people at least have the guts to talk to them first. These are real business with real employees who are building out the Bitcoin ecosystem. I have orders of magnitude more respect for people who have put their money, time and souls into Bitcoin ecosystem companies than for those who declare what is _tolerable_ from the sidelines.

This action is consistent with the divisive governance of this website. Previously a tweet has been used as justification to remove Trezor from the wallets page (#1564). Luckily that PR wasn't merged because reasonable heads prevailed.

From now on, I won't recommend using bitcoin.org. That's a shame because I think @harding and @wbnns do awesome work on the developer docs. I've contributed to that section in the past, but at this point I feel very uncomfortable having my name associated with this repo.

luke-jr commented:
> Did you speak to all of these companies before "denouncing" and delisting them? Do you know what their understanding of the NYA was when they signed up?

At this point, they've had plenty of time to publicly denounce 2X.

> Do you know what their customers' needs are?

If their customers' needs are exclusively non-Bitcoin, then there's no reason for a Bitcoin website to promote them.

> Do you understand what their responsibilities are to their investors, their employees and their customers?

Pretty sure fraud isn't among those responsibilities.

> Much more productive would be to issue a blanket warning to all Bitcoin users to understand what the hardfork in November means for their coins,

There is no hardfork in November, just a new altcoin launching with a plan to fraudulently pass itself off as Bitcoin.

> And anyway, who made @theymos and @Cobra-Bitcoin the moral arbiters of Bitcoin?

They are the owners of the website Bitcoin.org. As such, they can promote or not promote what they like with it.

> The previous PR (#1824) had NACKs from @wbnns, @ekerstein, @Mirobit, @nvk, @sunnankar and others.

None of who own the website. Note that @nvk withdrew his NACK after revisions were made, and also that some of the others were based on the specific format and way it was expressed (which AFAIK was revised multiple times, including the transition to this new PR).

I don't necessarily agree with the wording as-is, but I don't see some huge injustice going on here.

harding commented:
> Much more productive would be to issue a blanket warning to all Bitcoin users to understand what the hardfork in November means for their coins, and strongly advise them to contact all of their Bitcoin service providers and demand that their Bitcoin continue to be available and named Bitcoin after November.

@jnewbery How would several thousand separate people each taking time out of their day to contact their service providers to ask exactly what we have asked for be more productive than us asking them all en masse?

The companies that were delisted and which we intend to publish an alert about were not targeted arbitrarily.  They signed a letter announcing their intention to "Activate a 2 MB hard fork within six months".  If they didn't know that a hard fork to an untested new max block size on a tight deadline without proven community support would harm Bitcoin users, then I'm not sure they're good stewards of user deposits in the first place.  I think we've done them a courtesy by giving them a chance to describe how they plan to safeguard user bitcoins rather than immediately delisting them when the statement came out per our two-year-old [written policy](https://bitcoin.org/en/posts/hard-fork-policy) and alerting users then about the risks.

Regarding your other points, you're right that life for Bitcoin.org contributors is greatly complicated by involvement in semi-political issues like hard forks.  Indeed, a few weeks ago, I joked with another contributor that we should simply just remove all external links and recommendations to end the noise.  But really I think the only solution to the hard fork drama that has gripped Bitcoin for the last several years is the solution Bitcoin.org set upon more than two years ago with our previous policy: we reject contentious hard forks.

It is the responsibility of a group proposing a hard fork to build consensus around the proposal, to get nearly everyone to agree to the new consensus rules and the conditions under which they'll be adopted.  Any attempt at circumventing that consensus building will be rejected by Bitcoin.org---and rejected loudly if necessary.

jnewbery commented:
@harding - I have enormous respect and gratitude for your work here and elsewhere. Communicating technical Bitcoin concepts to a general audience is a **really** important and difficult job, and I don't know anyone who does it better than you.

I also suspect that our hopes for Bitcoin are fairly well aligned.

> How would several thousand separate people each taking time out of their day to contact their service providers to ask exactly what we have asked for be more productive than us asking them all en masse?

Because what I hope we're doing is building a movement, and helping people realise that they have sovereignty over their own money. I'm not interested in a system where 'we' (whoever that we is) have to denounce on users' behalf. There will another contentious hard fork after 2x, and another after that. Do we set ourselves up to be the judges of what Bitcoin is for perpetuity, or do we try to strengthen and educate the community?

As for taking time out of their day - that's what we all signed up for when we got into Bitcoin.

> It is the responsibility of a group proposing a hard fork to build consensus around the proposal, to get nearly everyone to agree to the new consensus rules and the conditions under which they'll be adopted. Any attempt at circumventing that consensus building will be rejected by Bitcoin.org---and rejected loudly if necessary.

Yes, I 100% agree, and I'd also be very happy to stand behind any condemnation of 2x attempting to take the 'Bitcoin' name.

Perhaps my previous comment wasn't clear. I'm in no way advocating for 2x. I think the btc1 implementation is inexcusably reckless and dangerous and will cause great harm. But I don't think this is the proper way to go about opposing it:

- without actually speaking to the companies that have been singled out here, the worst assumptions are being made about their motivations and positions. I would much rather see the rhetoric on all sides softened and an attempt made to bring people back together. Perhaps you think that's impossible at this stage.
- this PR was opened and merged in a day. Why the rush? Why not wait to hear from the people who commented on the previous PR?

friendsofbitcoin commented:
@jnewbery While I respect you position, I would like to clarify there exists a wider context here. The NYA was signed on May 23, and many of us have been repeatedly begging for clarification from these companies for months only to be met with silence, ambiguity, or outright deception. If anything this merge is coming too late and the delays will cause more users loss of funds due to the small window of time we have to educate others. I have no problems if these companies wish to create altcoins or leave bitcoin but the reckless behavior the companies are exhibiting on their lack of clarity, lack of seeking consensus from their own clients, and lack of communication is being interpreted as an attempt of fraud by many.

gmaxwell commented:
There is another argument for reaching out to parties beyond avoiding confusion-- it creates an opportunity to correct their misunderstandings.  Without doing that, you cede control of the narrative they here to people like Jgarzik who are flying around to meet with these people in person.

kallewoof commented:
A lot of people have been doing a lot of reaching out for a long time. I myself have tried repeatedly to get the attention, especially of the Swedish and Japanese companies on the list (the Swedish company responded, saying they will "let the market decide", and the Japanese company (bit flyer) ignored my repeated queries).

I think putting them on a list to show the users to beware of them due to their unwillingness to make their stance clear on the issue is the only thing we can do.

harding commented:
> I'm not interested in a system where 'we' (whoever that we is) have to denounce on users' behalf.

I've noticed several people who are upset by the verb "denounce".  I chose that phrasing, and so I'm to blame for any problems it caused.  All I meant by it was that we plan to publish an alert that will warn users that NYA signees are planning to hard fork Bitcoin, and that to do so without enacting additional protections will likely lead to them losing their users' deposits of non-HF bitcoins.

If it were any other situation and you had a list of companies that was likely 40 days away from losing their users' bitcoin deposits, would you not take a similar course of action by (1) asking the companies to describe how they will protect user bitcoin deposits and (2) for any company that doesn't provide an adequete description of their safety plan, warning users to remove their bitcoins from those services?

To me, this is not a case of us deciding what is and what is not Bitcoin---although the owners of this domain (like all Bitcoin users) do need to decide what consensus rules they'll enforce with their nodes and, presumably, on their website---but a case of us rejecting attempts to change Bitcoin's rules without community consensus and warning users about the losses they can suffer from those who make the attempt anyway.

> without actually speaking to the companies that have been singled out here, the worst assumptions are being made about their motivations and positions.

Again, I remind you that we did not select these companies.  They selected themselves by signing the NYA.

I imagine you're familiar with the truism, "hope for the best, but plan for the worst".  I think that's what we're doing here.  I think I speak for all the supporters of this PR is saying that we would very hope to see all of the NYA signatories clarify their positions in way that obviates the need to publish the alert.  But, speaking for myself now, I think we have to plan for the case where they will use HF'd software, spend pre-HF UTXOs without replay protection, and accept post-HF UTXOs without replay, and by doing so wipe out their users' bitcoin deposits (likely quite quickly once someone discovers they can churn deposits to get coins on both chains).

> I would much rather see the rhetoric on all sides softened and an attempt made to bring people back together. Perhaps you think that's impossible at this stage.

Although plans can be changed, I think we are generally resolved to publish the alert warning users about bitcoin deposits held by NYA signatories.  If you can suggest ways to improve the phrasing and eliminate potential empty rhetoric, I think they will be well considered.  In addition, I invite you to criticize me privately (or publicly) if you see me writing empty rhetoric anywhere.

> this PR was opened and merged in a day. Why the rush? Why not wait to hear from the people who commented on the previous PR?

On this point, I support you.  I have occasionally been criticized myself of merging too soon, but I do think fast and under-reviewed merges have been used too often in recent months on Bitcoin.org.

Finally, your respect for me cannot exceed my respect for you.  I admire you all the more for your willingness to write your criticism of this PR.

jnewbery commented:
Thanks for your thoughtful response David. I appreciate you taking the time.

From what I'm hearing, we're very much in agreement in our attitude to 2x. It's contentious, the details of the implementation are rushed and reckless, it will certainly result in confusion and lost funds for users, and in great expense for businesses. I also agree that it's vital to see clarification from all Bitcoin companies (not just NYA signatories) on how they plan to protect users' Bitcoin and avoid confusion.

My objection is not to any of that, but to the language and approach. Without actually speaking to the CEOs of *all* of these companies *first*, this reeks of McCarthyism. Yes, they signed the NYA, but without speaking to them, you don't know what they thought they were signing up to, and what their stance is now. Thanks @kallewoof for reaching out to two of the companies on the list. That's very admirable. How many others have made genuine attempts to reach out to these companies and understand their positions?

It seems like I'm on the minority fork of this debate so I won't keep on posting, but I'll finish by urging you to actually speak to people at these companies if you have the chance. Some have been at the receiving end of a torrent of abuse and death threats because of this episode. That's completely unacceptable and we should try very hard to not add fuel to that fire.

I hate to see these bridges being burned for no reason. When the *real* enemies of Bitcoin make their move, we might realise that there's more that unites than divides us.

CosmicHemorroid commented:
@jnewbery I think it's dangerous for you to repeat this unless you have personal knowledge. They could be spreading rumors to make the opposition (us) look bad.

"...and death threats because of this episode."

jnewbery commented:
> it's dangerous for you to repeat this unless you have personal knowledge

I've spoken to the CEO of one of the NYA companies and he told me this. I have no reason to believe he was lying. It seems perfectly plausible given the invective I've seen from trolls on twitter and reddit.

This doesn't make us look bad because we've never encouraged trolling or personal attacks, but we can also understand that a more hostile environment gives cover to bad actors.

Cobra-Bitcoin commented:
> Yes, they signed the NYA, but without speaking to them, you don't know what they thought they were signing up to, and what their stance is now. 

@jnewbery It's not our business to contact the CEO's asking them what they thought they signed, what their stance is, whether they will change, what they ate for dinner, etc. We are not their friends, and we don't have time to engage in this pointless "reaching out". They have very publicly stated their intention to support the hard fork, and unless they publicly withdraw from the NYA, we are forced to warn our users about doing business with them and possible loss of funds. What these people say in private regarding the NYA holds no weight if they aren't prepared to withdraw publicly. They may tell you in private they're "not happy" with it but what does that matter if they're going to go ahead and fork anyway?

There should be no mercy for hostile actors trying to change our network and harm our money. The entire community needs to come together and resist them, and not in a polite quiet "respectful" way, we need to use every resource at our disposal to make sure they don't succeed. We aren't going to chase around emailing CEO's asking "what they thought they signed", these people know exactly what they've signed, and they had plenty of time to withdraw support once they realized this change would be contentious. I have huge respect for the people in the community who are out on social media _every_ day resisting these companies and the change they want to impose on us. Some are polite, some not so much, but god bless these people for defending their network and showing these businesses that the bitcoin community will not bend over to corporate control.

> Some have been at the receiving end of a torrent of abuse and death threats because of this episode. That's completely unacceptable and we should try very hard to not add fuel to that fire.

Don't attack people's money and you won't get death threats. Assassination markets were an important cypherpunk concept for a reason. When someone attacks an entire community's money, they shouldn't be surprised to find that some angry, mentally ill, and insane people within that community want to do them harm.

jnewbery commented:
> Assassination markets were an important cypherpunk concept for a reason. When someone attacks an entire community's money, they shouldn't be surprised to find that some angry, mentally ill, and insane people within that community want to do them harm.

This is exactly why I have no intention of being associated with this website.

CosmicHemorroid commented:
@jnewbery "I've spoken to the CEO of one of the NYA companies and he told me this." IMO let them speak for themselves. You have no personal knowledge it's true and could be actually helping them spread viscous rumors that are unfounded.

nvk commented:
This thread has become pointless. Can any admin lock/close it pleas.

harding commented:
> There should be no mercy for hostile actors trying to change our network and harm our money. [...] we need to use every resource at our disposal to make sure they don't succeed. [...] Don't attack people's money and you won't get death threats. Assassination markets were an important cypherpunk concept for a reason. When someone attacks an entire community's money, they shouldn't be surprised to find that some angry, mentally ill, and insane people within that community want to do them harm.

@Cobra-Bitcoin I also want nothing to do with your violent agitation.  I'll submit a PR to remove my name from the about-us page, and then I'm out.

Cobra-Bitcoin commented:
@harding You took two completely disjoint sections of my reply and put them side by side to make my post seem much more threatening and menacing than it actually is. Communicating over text isn't ideal, but nothing about what I said even comes close to violent agitation, though maybe you read it in a tone which was more harsh than what I imagined when writing it.

harding commented:
@Cobra-Bitcoin I agree that communication is hard; I write professionally, and I screw it up more often than not.

That said, I think it's probably time I stop getting involved in these arguments.  I haven't been contributing much here for several months, and I think it's unlikely that I'll start again, so I should leave discussions up to the active contributors.

Cobra-Bitcoin commented:
Locking this, and cleared up all the unnecessary arguments. I think we all got a bit heated and angry.
